recognition/include/pcl/recognition/face_detection/face_common.h

```diff
@@ -127,15 +127,21 @@ namespace pcl
         {
           int num_of_sub_nodes;
           stream.read (reinterpret_cast<char*> (&num_of_sub_nodes), sizeof(num_of_sub_nodes));
+          PCL_CHECK_IO_STREAM(stream, "num_of_sub_nodes");
 
-          if (num_of_sub_nodes > 0)
+          if (num_of_sub_nodes <= 0)
           {
-            feature.deserialize (stream);
-            stream.read (reinterpret_cast<char*> (&threshold), sizeof(threshold));
+            PCL_THROW_EXCEPTION (pcl::IOException, "Error! Number of sub nodes read from file is not positive!");
```

Comment:
This is a change in behavior from before. The earlier code didn't throw in this case and continued ahead